### PR TITLE
stellar-core: init at 0.5.1

### PR DESCRIPTION
**Pull request by chris-martin**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/altcoins/stellar-core.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+let
+  pname = "stellar-core";
+  version = "0.5.1";
+
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
+
+  src = fetchgit {
+    url = "https://github.com/stellar/stellar-core.git";
+    rev = "refs/tags/v${version}";
```
Comment (LnL7) [inline pkgs/applications/altcoins/stellar-core.nix]: I'm pretty sure you can just use `v${version}"`

File: pkgs/applications/altcoins/stellar-core.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    sha256 = "0ldw3qr0sajgam38z2w2iym0214ial6iahbzx3b965cw92n8n88z";
+    fetchSubmodules = true;
+    leaveDotGit = true;
+  };
+
+  buildInputs = [ autoconf automake libtool pkgconfig git ];
+
+  propagatedBuildInputs = [ bison flex postgresql ];
+
+  patches = [ ./stellar-core-dirty-version.patch ];
```
Comment (LnL7) [inline pkgs/applications/altcoins/stellar-core.nix]: Why the patch? Seems like you could just use substituteInPlace.

File: pkgs/applications/altcoins/stellar-core.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    # Everything needs to be staged in git because the build uses
+    # `git ls-files` to search for source files to compile.
+    git add .
+
+    ./autogen.sh
+  '';
+
+  meta = with stdenv.lib; {
+    description = ''
+      Implements the Stellar Consensus Protocol, a federated consensus protocol
```
Comment (LnL7) [inline pkgs/applications/altcoins/stellar-core.nix]: I don't think that the description is supposed to be a multiline string.

**Review by LnL7 — COMMENTED** [review]
just some minor stuff

File: pkgs/applications/altcoins/stellar-core.nix (hunk 1 above)
Comment (chris-martin) [inline pkgs/applications/altcoins/stellar-core.nix]: You can't; it fails with "Bad commit hash or bad reference."

File: pkgs/applications/altcoins/stellar-core.nix (hunk 2 above)
Comment (chris-martin) [inline pkgs/applications/altcoins/stellar-core.nix]: It bothers me that `substituteInPlace` doesn't fail if it doesn't match anything. If the makefile changes in a future version, the substitution could just stop working and likely no one would notice. The patch approach will fail fast, which seems preferable?

File: pkgs/applications/altcoins/stellar-core.nix (hunk 3 above)
Comment (chris-martin) [inline pkgs/applications/altcoins/stellar-core.nix]: Okay, updated.

File: pkgs/applications/altcoins/stellar-core.nix (hunk 1 above)
Comment (LnL7) [inline pkgs/applications/altcoins/stellar-core.nix]: Strange.

**Pull request merged**